files/sdlrwopsstreambuf.cc

```diff
@@ -0,0 +1,281 @@
+	return this;
+}
+
+SdlRwopsStreambuf* SdlRwopsStreambuf::close() {
+	if (!m_context) {
+		return nullptr;
+	}
+
+	std::unique_ptr<SDL_RWops, int (*)(SDL_RWops*)> h(m_context, SDL_RWclose);
+	auto                                            rt = this;
+	if (sync())
+		rt = nullptr;
+	if (SDL_RWclose(h.release()))
+		rt = nullptr;
+	m_context = nullptr;
+	setg(nullptr, nullptr, nullptr);
+	setp(nullptr, nullptr);
+
+	return rt;
+}
+
+typename SdlRwopsStreambuf::int_type SdlRwopsStreambuf::underflow() {
+	if (!m_context)
+		return traits_type::eof();
+
+	bool initial = false;
+	if (!(m_currentMode & std::ios_base::in)) {
+		setp(nullptr, nullptr);
+		setg(m_buffer, m_buffer + sizeof(m_buffer),
+			 m_buffer + sizeof(m_buffer));
+		m_currentMode = std::ios_base::in;
+		initial       = true;
+	}
+
+	char_type buf1;
+	if (gptr() == nullptr)
+		setg(&buf1, &buf1 + 1, &buf1 + 1);
+	const std::size_t unget_sz
+			= initial ? 0 : std::min<std::size_t>((egptr() - eback()) / 2, 4);
+	int_type c = traits_type::eof();
+	if (gptr() == egptr()) {
+		std::memmove(eback(), egptr() - unget_sz, unget_sz * sizeof(char_type));
+		std::size_t nmemb = egptr() - eback() - unget_sz;
+		nmemb             = SDL_RWread(m_context, eback() + unget_sz, 1, nmemb);
+		if (nmemb != 0) {
+			setg(eback(), eback() + unget_sz, eback() + unget_sz + nmemb);
+			c = traits_type::to_int_type(*gptr());
+		}
+	} else
+		c = traits_type::to_int_type(*gptr());
+	if (eback() == &buf1)
+		setg(nullptr, nullptr, nullptr);
+	return c;
+}
+
+typename SdlRwopsStreambuf::int_type SdlRwopsStreambuf::pbackfail(int_type c) {
+	if (m_context && eback() < gptr()) {
+		if (traits_type::eq_int_type(c, traits_type::eof())) {
+			gbump(-1);
+			return traits_type::not_eof(c);
+		}
+		if ((m_openMode & std::ios_base::out)
+			|| traits_type::eq(traits_type::to_char_type(c), gptr()[-1])) {
+			gbump(-1);
+			*gptr() = traits_type::to_char_type(c);
+			return c;
+		}
+	}
+	return traits_type::eof();
+}
+
+typename SdlRwopsStreambuf::int_type SdlRwopsStreambuf::overflow(int_type c) {
+	if (!m_context)
+		return traits_type::eof();
+
+	if (!(m_currentMode & std::ios_base::out)) {
+		setg(nullptr, nullptr, nullptr);
+		setp(nullptr, nullptr);
+		m_currentMode = std::ios_base::out;
+	}
+
+	char_type  buf1;
+	char_type* pb_save  = pbase();
+	char_type* epb_save = epptr();
+	if (!traits_type::eq_int_type(c, traits_type::eof())) {
+		if (pptr() == nullptr)
+			setp(&buf1, &buf1 + 1);
+		*pptr() = traits_type::to_char_type(c);
+		pbump(1);
+	}
+	if (pptr() != pbase()) {
+		std::size_t nmemb = static_cast<std::size_t>(pptr() - pbase());
+		if (SDL_RWwrite(m_context, pbase(), sizeof(char_type), nmemb) != nmemb)
+			return traits_type::eof();
+		setp(pb_save, epb_save);
+	}
+	return traits_type::not_eof(c);
+}
+
+typename SdlRwopsStreambuf::pos_type SdlRwopsStreambuf::seekoff(
+		off_type off, std::ios_base::seekdir dir, std::ios_base::openmode) {
+	int width = 1;
+	if (!m_context || (width <= 0 && off != 0) || sync())
```

Comment:
We already know width is 1 here, so the check `width <= 0` is always false (see [LGTM 1](https://lgtm.com/projects/g/exult/exult/snapshot/a28cf7660fd554c904f32373c65983cdbc5fb585/files/files/sdlrwopsstreambuf.cc#xf56c03303f434649:1).